lib/internal/http2/core.js

```diff
@@ -1128,60 +1266,45 @@ function afterDoStreamWrite(status, handle, req) {
 
   if (session !== undefined)
     session[kState].writeQueueSize -= bytes;
-
   if (typeof req.callback === 'function')
-    req.callback();
+    req.callback(null);
   req.handle = undefined;
 }
 
 function onHandleFinish() {
+  const handle = this[kHandle];
   if (this[kID] === undefined) {
     this.once('ready', onHandleFinish);
-  } else {
-    const handle = this[kHandle];
-    if (handle !== undefined) {
-      const req = new ShutdownWrap();
-      req.oncomplete = () => {};
-      req.handle = handle;
-      handle.shutdown(req);
-    }
+  } else if (handle !== undefined) {
+    const req = new ShutdownWrap();
+    req.oncomplete = () => {};
+    req.handle = handle;
+    handle.shutdown(req);
   }
 }
 
-function onSessionClose(hadError, code) {
-  abort(this);
-  this.push(null);  // Close the readable side
-  this.end();       // Close the writable side
-}
-
 function streamOnResume() {
-  if (this[kID] === undefined) {
-    this.once('ready', streamOnResume);
-    return;
-  }
-  this[kHandle].readStart();
+  if (!this.destroyed && !this.pending)
+    this[kHandle].readStart();
 }
 
 function streamOnPause() {
-  this[kHandle].readStop();
-}
-
-function handleFlushData(handle) {
-  handle.flushData();
+  // if (!this.destroyed && !this.pending)
+  //   this[kHandle].readStop();
```

Comment:
why is this commented out?

Reply:
@jasnell stumbling over this again … do you remember why? I’m looking into some refactors on the read side of http2 streams and I guess I’ll find out myself soon, but if you do remember that would be helpful :)